src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -315,13 +320,20 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   auto timeout_point = ToTimeoutPoint(timeout_ms);
   bool locked = false;
   bool expired = false;
-  do {
-    RAY_RETURN_NOT_OK(object->header->CheckHasError());
-    // The channel is still open. This lock ensures that there is only one reader
-    // at a time. The lock is released in `ReadRelease()`.
-    locked = channel->lock->try_lock();
-    expired = timeout_point && std::chrono::steady_clock::now() >= *timeout_point;
-  } while (!locked && !expired);
+  {
+    std::signal(SIGINT, SignalHandler);
+    do {
+      RAY_RETURN_NOT_OK(object->header->CheckHasError());
+      // The channel is still open. This lock ensures that there is only one reader
+      // at a time. The lock is released in `ReadRelease()`.
+      locked = channel->lock->try_lock();
+      expired = timeout_point && std::chrono::steady_clock::now() >= *timeout_point;
+    } while (!locked && !expired && signal_received != -1);
+  }
+  if (signal_received != -1) {
+    RAY_LOG(ERROR) << "Interrupted by signal";
+    return Status::Interrupted("Interrupted by signal");
```

Comment:
We should print out what the signal is.

Also, can you show an example in the description what the final exception looks like?

Reply:
added in description